### ISSUE TITLE
bootstrap.iso for rhel 7.5 having some issues

### ISSUE THREAD

**Issue by malikkal** [vmware/vic#8569]
Summary
bootstrap.iso file builds successfully and containers could startup however, rngd is not running and iptables is incorrect.

vSphere and vCenter Server version
vSphere 6.7u2

VIC version
VIC Version: 1.5.3

iptables using wrapper
![image](https://user-images.githubusercontent.com/16984768/60482451-26345e00-9cc4-11e9-8211-d9c4d77ac654.png)

iptables without wrapper (see ruleset below). A ping to gateway would result in "sendmsg: Operation not permitted".
![image](https://user-images.githubusercontent.com/16984768/60482490-5845c000-9cc4-11e9-891a-e79106d456bf.png)

rngd not running; attempting to test start manually, resulted in library error. additional package required during build?
![image](https://user-images.githubusercontent.com/16984768/60482571-a4910000-9cc4-11e9-8f43-1c805faa0863.png)

missing library in /.tether/lib64
![image](https://user-images.githubusercontent.com/16984768/60482609-d4400800-9cc4-11e9-8725-88d4a9a83205.png)

Also noticed that base.sh and staging.sh files are not required for centos7/rhel7 builds. if they are present the build would fail.

@aviratna
@hickeng

**Comment by hickeng** [comment]
> Also noticed that base.sh and staging.sh files are not required for centos7/rhel7 builds. if they are present the build would fail.

My recommendation would be to build with debug to get more verbosity about why the base.sh and staging.sh files cause the build to fail (if using the docker container approach, make sure to add `-e DEBUG=1`):
```
export DEBUG=1
export REPO=rhel-7.5
make bootstrap-custom
```

If you've copied base.sh and staging.sh from old RHEL releases then they may well need revision before they'll work.
`iptables` is sensitive to having all of it's component shared libraries in the locations that version expects.
`rngd` I recall having some issues with to begin with which is why `--no-tpm` is there, but also why photon-2.0, for example, used `haveged` instead.

The base.sh, staging.sh, and init.sh are to allow for significant customization of the libraries and packages that go into the image - I think you'll need to make revisions for 7.5

You _could_ try starting using those files from photon-2.0 for example where there's a different set of libraries and utils.

Basically the answer is:
a. find out what libraries need to be present where for iptables
b. find out what RNG works on RHEL7.5 and add that.

Sorry I'm unable to actually provide revised versions of those files at this time. @DanielXiao may be able to assist - he was involved in the initial custom bootstrap work.

**Comment by hickeng** [comment]
Possible approach to determining what's missing for iptables:

Run the same commands used in the container (see tether.debug with debug for verbose dump) and run those commands in a regular 7.5:
```bash
LD_DEBUG=libs LD_DEBUG_OUTPUT=/tmp/iptables_dynamic_dependencies.out iptables ...
```
That will give a giant list of rtld output but means you should be able to see what's expected, and then compare with what that LD_DEBUG shows in the container image.

**Comment by malikkal** [comment]
Thanks @hickeng 

For photon, The samples provided under https://github.com/vmware/vic/tree/master/isos/base/repos/photon-2.0 doesn't have stanging.sh.

**Comment by DanielXiao** [comment]
@malikkal Did you use sample https://github.com/vmware/vic/tree/master/isos/base/repos/centos-7 to build bootstrap.iso for rhel 7.5? I need more information to reproduce this issue:
1. RHEL Kernel bits or similar one from CentOS.
2. Scripts you used to build bootstrap.iso.

**Comment by malikkal** [comment]
Thank you @DanielXiao. I have used the sample from https://github.com/vmware/vic/tree/master/isos/base/repos/centos-7. However, since the sample from that folder didn't have base.sh and staging.sh, I assume they are not required. Also, if you place them in 7.x copied over from 6.9 samples, the build would fail.

I tried with files from photon2.0 samples. Although the build passes, I get the following error.
![image](https://user-images.githubusercontent.com/16984768/60510027-03746a80-9d01-11e9-9149-04fe80ab6714.png)

obviously, the /.tether/lib doesn't have the required libraries...

**Comment by malikkal** [comment]
I ran it with and without debug. Please see my putty log. I don't think the init.sh under repos folder is being called.

[putty.log](https://github.com/vmware/vic/files/3351033/putty.log)

**Comment by DanielXiao** [comment]
I am looking at it now.

**Comment by aviratna** [comment]
@DanielXiao  Can you please let us know if there are any updates.

@malikkal

**Comment by aviratna** [comment]
@YanzhaoLi : Can you please let us know if there are any updates. Earlier We were following up with Daniel. This issue is critical for us as we have deployment waiting for this issue to be resolve.

**Comment by YanzhaoLi** [comment]
@aviratna @malikkal 
Please use the centos-7 configuration that I made in my pr #8574. I've tested that it works.
